src/nnet/nnet-parametric-relu.h

```diff
@@ -0,0 +1,284 @@
+    WriteBasicType(os, binary, clip_gradient_);
+
+    // ParametricRelu scaling parameters
+
+    if (!binary) os << "\n";
+     max_alpha_.Write(os, binary);
+     min_beta_.Write(os, binary);
+  }
+
+  int32 NumParams() const {
+    return max_alpha_.Dim() + min_beta_.Dim();
+  }
+
+  void GetGradient(VectorBase<BaseFloat>* gradient) const {
+    KALDI_ASSERT(gradient->Dim() == NumParams());
+    int32 max_alpha_num_elem = max_alpha_.Dim();
+    int32 min_beta_num_elem = min_beta_.Dim();
+    gradient->Range(0, max_alpha_num_elem).CopyFromVec(Vector<BaseFloat>(max_alpha_corr_));
+    gradient->Range(max_alpha_num_elem, min_beta_num_elem).CopyFromVec(Vector<BaseFloat>(min_beta_corr_));
+  }
+
+  void GetParams(VectorBase<BaseFloat>* params) const {
+    KALDI_ASSERT(params->Dim() == NumParams());
+    int32 max_alpha_num_elem = max_alpha_.Dim();
+    int32 min_beta_num_elem = min_beta_.Dim();
+    params->Range(0, max_alpha_num_elem).CopyFromVec(Vector<BaseFloat>(max_alpha_));
+    params->Range(max_alpha_num_elem, min_beta_num_elem).CopyFromVec(Vector<BaseFloat>(min_beta_));
+  }
+
+  void SetParams(const VectorBase<BaseFloat>& params) {
+    KALDI_ASSERT(params.Dim() == NumParams());
+    int32 max_alpha_num_elem = max_alpha_.Dim();
+    int32 min_beta_num_elem = min_beta_.Dim();
+    max_alpha_.CopyFromVec(params.Range(0, max_alpha_num_elem));
+    min_beta_.CopyFromVec(params.Range(max_alpha_num_elem, min_beta_num_elem));
+  }
+
+  std::string Info() const {
+    return std::string("\n  max_alpha") +
+      MomentStatistics(max_alpha_) +
+      ", max-lr-coef " + ToString(max_learn_rate_coef_) +
+      "\n  min_beta" + MomentStatistics(min_beta_) +
+      ", min-lr-coef " + ToString(min_learn_rate_coef_);
+  }
+  std::string InfoGradient() const {
+    return std::string("\n  max_alpha_grad") +
+      MomentStatistics(max_alpha_corr_) +
+      ", max-lr-coef " + ToString(max_learn_rate_coef_) +
+      "\n  min_beta_grad" + MomentStatistics(min_beta_corr_) +
+      ", min-lr-coef " + ToString(min_learn_rate_coef_);
+  }
+
+  void PropagateFnc(const CuMatrixBase<BaseFloat> &in,
+                    CuMatrixBase<BaseFloat> *out) {
+    out->ParametricRelu(in, max_alpha_, min_beta_);
+  }
+
+  void BackpropagateFnc(const CuMatrixBase<BaseFloat> &in,
+                        const CuMatrixBase<BaseFloat> &out,
+                        const CuMatrixBase<BaseFloat> &out_diff,
+                        CuMatrixBase<BaseFloat> *in_diff) {
+    // multiply error derivative by activations (alpha, beta)
+    // ((dE/da)*w) === out_diff, f(y) == out,
+    //  (out > 0 ) ? out_diff * alpha : out_diff * beta
+    in_diff->DiffParametricRelu(out, out_diff, max_alpha_, min_beta_);
+  }
+
+
+  void Update(const CuMatrixBase<BaseFloat> &input,
+              const CuMatrixBase<BaseFloat> &diff) {
+    // we use following hyperparameters from the option class
+    const BaseFloat alr = opts_.learn_rate * max_learn_rate_coef_;
```

Comment:
It would be better to replace `max_learn_rate_coef_` and `min_learn_rate_coef_` by the variables `bias_learn_rate_coef` (alphas) and `learn_rate_coef_` (betas) from the interface `UpdatableComponent::`. These can be adjusted externally by the tool `nnet-set-learnrate`.